### PR TITLE
Fix Tests

### PR DESCRIPTION
**Pull request by grahamwren**
Looks like my formatter moved your macros a little, sorry about that

# More Importantly:

## Remove tests for Object and String
Your tests for `Object` and `String` produce additional requirements for the interface of `Object` and `String` which are not relevant to your interface or tests for `Map`. The assignment does not ask you to provide tests for `Object` and `String` just for your chosen interface. Because of this and to ease integration with others' code, I am deleting these tests.

## Add `;` to one of you macro calls
This just makes it compile.

### CODE REVIEW

**Review by jjones-99 — CHANGES_REQUESTED** [review]
Re: the macro semi-colon and the formatting stuff, that's all okay.

Re: deleting the tests for `Object` and `String`, that should **not** be done. Regardless of what the focus of the assignment is (in this case, `Map`) you should **always** test all of your code. To reference the assignment description, "Untested code is not something your company will ever use." The test suite should be "exhaustive" and cover all use cases.

Re: having extra methods and tests that might not seem necessary in this context (the `String` `concat()` method, for instance), this spec is in line with the standardized `String` and `Object` specs shared on [Piazza @345](https://piazza.com/class/k51bluky59n2jr?cid=345) by Chase Bishop (see [repo](https://github.com/chasebish/cwc_object_string)). We were asked in #6 to use that spec and some of the specs we received include that functionality. In order to align with the other teams, we'll leave those methods there. And in order to test that code, the corresponding tests must be included.

_Even if some of the other specs you received don't require those methods, implementing them shouldn't cause you problems. These tests are already written and the methods themselves are very quick to implement._

### TL;DR  - Deleting `String` and `Object` tests is rejected

Tests should always be exhaustive regardless of what you're asked to implement. If your own tests or the tests you received from other spec pairs don't cover at least all public use cases, that's something you should probably push to add.

_I've done some searching on Piazza and the assignment description and haven't found any counterpoints to this. If you find anything and link it to us, we can discuss more._

**Comment by jsxtam** [comment]
I agree with @jjones-99 that we should still test the `Object` and `String` classes. The tests we created for these classes cover all public use cases for the spec we are using. Even though some of these methods will probably not be relevant to you implementing `Map`, the functionality for these classes should still exist regardless. Implementing this now should not be too difficult and may potentially be useful for later assignments if done successfully for this one.

**Review by jsxtam — COMMENTED** [review]
I agree with @jjones-99 that we should still test the `Object` and `String` classes. The tests we created for these classes cover all public use cases for the spec we are using. Even though some of these methods will probably not be relevant to you implementing `Map`, the functionality for these classes should still exist regardless. Implementing this now should not be too difficult and may potentially be useful for later assignments if done successfully for this one.

(was formerly a non-review comment)

**Comment by grahamwren** [comment]
Due to changes in the assignment which previously asked us to write our own tests for classes other than the relevant interface, I will re-add most tests for Object and String.
However, re this PR https://github.com/chasebish/cwc_object_string/pull/8, I'd like to still remove the test for `concat`/`size` because the interface was poorly designed and will change. Thoughts?

**Comment by jjones-99** [comment]
Based on the PR you linked, it looks like there are going to be changes made to the signatures for `concat` and `size`. However, these changes won't be finalized and merged to `master` until after the Friday night deadline.

While I agree with the changes that will be made, they haven't been made yet. Thus, I think the tests should remain.

Think about it like this: I'm developing software for a public deployment in a week, and one of my interfaces is used by some of the other teams on the project. Some of the teams met over lunch and chatted about making some agreeable changes to that interface. However, because of the looming deadline—and all of the teams that didn't come to lunch with us and don't know about the incoming changes yet—we plan to update that interface after the public deployment, to be deployed next time.

When I write tests, I want to make sure next week's deployment is suitable for the public and test based on the current interface. I don't assume the changes we _want_ to make have been made.

After next week's deployment—and once the changes have been made and all teams are aware of them—the tests can be updated to match.

### TL;DR - Tests should still remain.

The tests for this assignment should exhaustively align with the interface for this assignment. If and when the interface changes, the tests can and should change to match.

**Comment by grahamwren** [comment]
Asked in class just now, Jan responded you should not have written tests for Object and String. Instead, "you should have only tested the interface you designed with as few dependencies as possible."
https://piazza.com/class/k51bluky59n2jr?cid=305

**Comment by jjones-99** [comment]
All righty. In that case, makes sense.
